auto_backup/models/db_backup.py

```diff
@@ -240,10 +240,10 @@ def cleanup(self):
                 oldest = self.filename(now - timedelta(days=rec.days_to_keep))
 
                 if rec.method == "local":
-                    for name in iglob(os.path.join(rec.folder,
-                                                   "*.dump.zip")):
-                        if os.path.basename(name) < oldest:
-                            os.unlink(name)
+                    for name in sorted(
+                            iglob(os.path.join(rec.folder, "*.dump.zip"))
+                    )[:-rec.days_to_keep]:
+                        os.unlink(name)
 
                 elif rec.method == "sftp":
                     with rec.sftp_connection() as remote:
```

Comment:
The update affects also remote backups. Could you work on this part too ?

Reply:
done, I also added some (incomplete) tests for the remote case